website/server/middlewares/testServerMode.js

```diff
@@ -0,0 +1,25 @@
+import nconf from 'nconf';
+import {model as User} from '../models/user';
+
+const IS_TEST_SERVER = nconf.get('IS_TEST_SERVER');
+
+module.exports = function maintenanceMode (req, res, next) {
+  if (IS_TEST_SERVER !== true) return next();
+  const userId = req.header('x-api-user');
```

Comment:
I believe they worked for me, but I can check again.

Reply:
So I checked and they work because if no user is found line 17 here is never reached so it goes to the next directly after. So if you pass a wrong user id or no user id at all the check is skipped.

Now that I think about it there's a security issue here because we're just checking the user id which is publicly avaible so you just have to copy the user id of an admin and you'll be able to access the test server freely

Reply:
But shouldn't that then fail the actual authentication? So that while this test might be bypassed, the user could still not perform any actions since the server will then deny access because the api key won't match. But I can also add a check for the key here.